### PR TITLE
WAPP-1: add default styled-component and redux examples

### LINKED ISSUE #1: [WAPP-1] As a developer I want to set up project (ReactJS, Redux, Typescrypt, styled-components)
Set up the project using: 
 * TypeScript
 * React
 * Redux
 * Styled-components

### PR DESCRIPTION
**Pull request by anikitosiknik**
Screenshot 2021-09-03 at 14 29 42

### CODE REVIEW

File: src/StyledTest.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import styled from 'styled-components'
+
+export default function StyledTest () {
+
+    const Title = styled.h1`
```
Comment (IlyaRucavitcyn) [inline src/StyledTest.tsx]: Styling should not be inside components functions. Also please fix formatting here.

**Comment by timmalahov** [comment]
@anikitosiknik Do we really need those test components of yours in a master branch?

File: src/App.tsx [hunk 2]
```diff
@@ -2,10 +2,23 @@ import React from 'react';
...
+import { Provider } from 'react-redux';
+import { store } from './store/store';
+import ReduxTest from './ReduTest';
+import StyledTest from './StyledTest';
+
+
 function App() {
+
+
+
```
Comment (timmalahov) [inline src/App.tsx]: What's with all that whitespace?

File: src/store/counterSlice.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import { RootState } from './store'
+
+interface CounterState {
```
Comment (timmalahov) [inline src/store/counterSlice.ts]: Please, move interfaces to a separate files

File: src/store/hooks.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux'
+import type { RootState, AppDispatch } from './store'
+
+export const useAppDispatch = () => useDispatch<AppDispatch>()
```
Comment (IlyaRucavitcyn) [inline src/store/hooks.ts]: Can you explain why we need to create separate file and in fact reimport `useDispatch` and `useSelector`?

File: src/store/store.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { configureStore } from "@reduxjs/toolkit"
+import counterSlice from "./counterSlice"
+
+export const store = configureStore({
+  reducer: {
+    counter: counterSlice
+  },
+})
+
+export type RootState = ReturnType<typeof store.getState>
```
Comment (IlyaRucavitcyn) [inline src/store/store.ts]: Can you give me a few more details here - why we do it?

**Comment by IlyaRucavitcyn** [comment]
Please take a closer look at your formatting setup - a lot of end  line issues : 
image

**Review by IlyaRucavitcyn — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 6]
```diff
@@ -2,10 +2,23 @@ import React from 'react';
...
   return (
+    <Provider store={store}>
+      
     <div className="App">
       <header className="App-header">
+        <StyledTest/>
+        <ReduxTest/>
```
Comment (timmalahov) [inline src/App.tsx]: I don't think we need these things in a master branch

File: src/App.tsx [hunk 7]
```diff
@@ -2,10 +2,23 @@ import React from 'react';
...
+import StyledTest from './StyledTest';
+
+
 function App() {
+
+
+
   return (
+    <Provider store={store}>
+
```
Comment (timmalahov) [inline src/App.tsx]: Please fix the formatting here

File: src/store/counterSlice.ts (hunk 3 above)
Comment (timmalahov) [reply src/store/counterSlice.ts]: Also give them more appropriate name, e.g. you can use *Interface postfix.

File: src/store/store.ts (hunk 5 above)
Comment (timmalahov) [inline src/store/store.ts]: Correct me if I'm wrong, but wouldn't it be a function here?

**Comment by timmalahov** [comment]
Could you please use a more structured approach for the files and components in the future?

For example you can create a "Components" directory and create a separate folder for each component and it's surroundings in it.

For something more complex and store-related you can use "modules" directory instead.

File: src/StyledTest.tsx (hunk 1 above)
Comment (anikitosiknik) [reply src/StyledTest.tsx]: this component was removed as unnecessary

File: src/store/hooks.ts (hunk 4 above)
Comment (anikitosiknik) [reply src/store/hooks.ts]: this trick was provided by redux toolkit documentation for comfortable using types
https://redux-toolkit.js.org/tutorials/typescript#define-typed-hooks

File: src/store/store.ts (hunk 5 above)
Comment (anikitosiknik) [reply src/store/store.ts]: the same as above, separate types for handy using
https://redux-toolkit.js.org/tutorials/typescript#define-root-state-and-dispatch-types

File: src/App.tsx (hunk 2 above)
Comment (anikitosiknik) [reply src/App.tsx]: configured prettier, now it should be ok

File: src/App.tsx (hunk 7 above)
Comment (anikitosiknik) [reply src/App.tsx]: formatting fixed

File: src/App.tsx (hunk 6 above)
Comment (anikitosiknik) [reply src/App.tsx]: deleted

File: src/store/store.ts (hunk 5 above)
Comment (anikitosiknik) [reply src/store/store.ts]: no, it's a type of state

File: src/store/counterSlice.ts (hunk 3 above)
Comment (anikitosiknik) [reply src/store/counterSlice.ts]: this component was removed as unnecessary, I'll do all these modifications next time

File: src/App.tsx [hunk 8]
```diff
@@ -1,25 +1,18 @@
...
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <Provider store={store}>
+      <Main></Main>
```
Comment (IlyaRucavitcyn) [inline src/App.tsx:14]: In case of content absence we need to use self closing tags

**Review by IlyaRucavitcyn — APPROVED** [review]

**Pull request merged**